src/polygon/src/polygon.cpp

```diff
@@ -20,7 +20,7 @@
 #endif
 #include <list>
 #include <map>
-#if !__APPLE__
+#if __has_include(<memory_resource>)
```

Comment:
I ran into a problem where `__APPLE__` is not defined for a WASM build even on my Mac. However, it looks like C++17 has a much better solution. There are probably some other places we should use this too.

Reply:
Interesting. I somehow remember that another issue I encounter is the lack of support for pstl, that is why I put `__APPLE__` to those macros using PSTL. I think the `<execution>` header is there, but it does not contain the required definitions.

Reply:
Thinking about it, `__APPLE__` is not defined for WASM build sounds fine, as that is cross compilation and the target platform is indeed not apple. It is interesting that it builds correctly for me, as emscripten is using clang, which should not have `<memory_resource>` as well.

Anyway, this looks fine to me.